svelte/src/lib/utils/pagination.ts

```diff
@@ -0,0 +1,66 @@
+const VIEWABLE_PAGES = 9;
+
+export interface PaginationState {
+  currentPage: number;
+  currentPageStart: number;
+  currentPageEnd: number;
+  availablePages: number;
+  nextPage: number;
+  prevPage: number;
+  pages: number[];
+  maxPages?: number;
+}
+
+export function calculatePagination(
+  currentPage: number,
+  perPage: number,
+  totalItems: number,
+  maxPages?: number,
+): PaginationState {
+  let currentPageStart = totalItems === 0 ? 0 : (currentPage - 1) * perPage + 1;
+  let currentPageEnd = Math.min(currentPage * perPage, totalItems);
+
+  let calculatedPages = Math.ceil(totalItems / perPage || 1);
```

Comment:
This would result in an `infinite` value if `perPage` is `0`.

Reply:
same as before: https://github.com/rust-lang/crates.io/blob/62209ecba038bfaae97edcf514aea85a9e55f92c/app/utils/pagination.js#L19

😅